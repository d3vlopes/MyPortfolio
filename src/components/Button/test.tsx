import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
