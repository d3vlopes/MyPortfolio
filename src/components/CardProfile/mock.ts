import { CardProfileProps } from '.'

export default {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul, RS',
  liking: 'Programação, design, música, séries e filmes.',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
} as CardProfileProps
