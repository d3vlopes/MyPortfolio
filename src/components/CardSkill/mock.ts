import { CardSkillProps } from '.'

export default [
  {
    icon: '/img/icons/sword.svg',
    title: 'Desafios',
    description:
      'Estou sempre em busca de novos desafios que me tirem do meu porto seguro.',
  },
  {
    icon: '/img/icons/target-arrow.svg',
    title: 'Persistência',
    description: `Não desanimo em frente aos obstáculos, quanto mais
      difícil, melhor a satisfação.`,
  },
  {
    icon: '/img/icons/lightbulb.svg',
    title: 'Agilidade',
    description: `Tenho facilidade para pensar em possíveis soluções para problemas.`,
  },
  {
    icon: '/img/icons/trending-up.svg',
    title: 'Evolução',
    description: `Gosto muito de aprender coisas novas e estar sempre sempre evoluindo.`,
  },
] as CardSkillProps[]
