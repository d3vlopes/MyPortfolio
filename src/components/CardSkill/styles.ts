import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: ${theme.colors.complementary.main};
    border-radius: ${theme.border.radius.large};
    width: min(29rem, 100%);
    padding: 3rem;
    transition: all 0.3s ease-in-out;
    height: min(29.6rem, 33rem);

    &:hover {
      transform: scale(1.1);
      background: ${theme.colors.complementary.dark};
      box-shadow: 0px 10px 15px rgba(202, 213, 217, 0.1);

      ${Title}::after {
        content: '';
        position: absolute;
        left: 0;
        bottom: -0.8rem;
        margin-bottom: ${theme.spacings.xsmall};
        width: 100%;
        display: inline-flex;
        border: 0.2rem solid ${theme.colors.primary.main};
        margin-bottom: ${theme.spacings.xxsmall};
      }
    }
  `}
`

export const Icon = styled.img`
  max-width: min(6rem, 100%);
`

export const Title = styled.h3`
  ${({ theme }) => css`
    position: relative;
    font-family: ${theme.font.heading};
    font-size: ${theme.font.sizes.large};
    color: ${theme.colors.text.dark};
    padding: ${theme.spacings.xsmall} 0 ${theme.spacings.xxsmall} 0;
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    line-height: 150%;
    text-align: center;
    color: ${theme.colors.text.dark};
    padding: ${theme.spacings.xsmall};
  `}
`
