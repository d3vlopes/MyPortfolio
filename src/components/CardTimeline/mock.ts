import { CardTimelineProps } from '.'

export default [
  {
    title: 'O início',
    content: `<p>Desde de muito cedo sempre gostei muito de tecnologia,
    como video game, computador, celular,
    mais nunca tinha tido interesse de querer saber como tudo aquilo funcionava.</p>
    <p>Meu primeiro contato com programação,
    foi durante o curso onde me formei Técnico em Informática pelo Senac.</p>
    <p>Foi algo mágico, eu digo que foi “amor a primeira linha de código”,
    eu me apaixonei por como tudo aquilo funcionava,  como cada tag fazia algo diferente,
    como a gente podia transformar o visual de um documento HTML, com a adição do CSS.
    Eu fiquei maravilhado todo aquele mundo e
    a infinidade de possibilidades que eu tinha descoberto.</p>
    `,
  },
  {
    title: 'InFome',
    content: `<p>Durante o curso, participei de alguns projetos com meus colegas,
    o principal deles foi a criação do InFome,
    um projeto com a ideia de divulgar lugares pouco conhecidos da cidade,
    mostrando informações, imagens, localizção,
    incentivando o usuário ir até o lugar divulgado.</p>
    <p>O projeto chegou a ser pré-incubado pela ITUNISC,
    que incentiva micro e pequenas empresas tecnologicamente inovadoras.</p>
    <p>Fui responsável por todo o desenvolvimento da aplicação,
    backend e frontend, junto com a cordenação do meu professor.</p>
    <p>Infelizmente o projeto não foi em frente,
    mais todo o aprendizado e vivência que tive, foi algo que me agregou muito.</p>
    `,
  },
  {
    title: 'Primeiro emprego',
    content: `<p>Após a formação, tive a oportunidade de trabalhar em uma
    agência de desenvolvimento aqui da cidade a Empório Adamantis.
    Foi como um sonho realizado estar ali dentro e poder ver de perto todo o processo
    de criação, desde do briefing, design até o desenvolvimento.</p>
    <p>Trabalhei lá como desenvolvedor Full Stack, atuava tanto no backend,
    como no frontend, criando novas aplicações ou fazendo alterações em alguma
    que já estava em produção quando necessário.</p>
    <p>Fiquei lá um período curto, mais foi uma experiência muito importante para mim,
    me ajudou a se transformar no profissional que sou hoje.</p>
    `,
  },
  {
    title: 'Frontend e Design',
    content: `<p>Depois da saida da Empório, foquei em estudar bastante,
    aprender novas tecnologias e me preparar bem para voltar ao mercado de trabalho.</p>
    <p>Foi ai que decidi me especializar no frontend,
    que era a parte do desenolvimento que eu mais gostava de trabalhar.</p>
    <p>E essa escolha de se especializar apenas em uma área do desenvolvimento
    também me ajudou a poder fazer outra coisa que eu amava,
    que era trabalhar com design e criação de interfaces.</p>
    <p>Eu sempre gostei muito de criar coisas,
    desenhar e era algo que eu tinha muita vontade de estudar e me especializar.</p>
    <p>Como os profissionais de UI e Frontend trabalham próxmos,
    foi perfeito conseguir juntar duas coisas que eu amo e trabalhar com elas juntas.</p>
    `,
  },
] as CardTimelineProps[]
