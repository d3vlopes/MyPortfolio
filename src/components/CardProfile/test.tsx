import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
