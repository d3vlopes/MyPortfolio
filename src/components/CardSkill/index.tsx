import * as S from './styles'

export type CardSkillProps = {
  icon: string
  title: string
  description: string
}

const CardSkill = ({ icon, title, description }: CardSkillProps) => {
  return (
    <S.Wrapper>
      <S.Icon src={icon} alt={title} />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
    </S.Wrapper>
  )
}

export default CardSkill
