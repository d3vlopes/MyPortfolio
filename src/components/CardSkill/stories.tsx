import { Story, Meta } from '@storybook/react'

import CardSkill, { CardSkillProps } from '.'
import mock from './mock'

export default {
  title: 'Components/CardSkill',
  component: CardSkill,
  args: {
    icon: mock[0].icon,
    title: mock[0].title,
    description: mock[0].description,
  },
  parameters: {
    layout: 'centered',
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/KIQWZ47GDrrNCIxysbEqOA/My-Portfolio?node-id=874%3A1072',
    },
  },
} as Meta<CardSkillProps>

export const Default: Story<CardSkillProps> = (args) => {
  return (
    <div style={{ display: 'flex', gap: '3.2rem', justifyContent: 'center' }}>
      <CardSkill {...args} />
    </div>
  )
}
