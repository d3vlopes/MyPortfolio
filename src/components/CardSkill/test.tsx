import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Sword } from '@styled-icons/remix-fill/Sword'

import CardSkill from '.'
import mock from './mock'

const { getByTestId, getByRole, getByText } = screen

describe('<CardSkill />', () => {
  it('should render CardSkill', () => {
    renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    const icon = getByTestId('icon')
    const title = getByRole('heading', { name: /desafios/i })
    const description = getByText(
      /Estou sempre em busca de novos desafios que me tirem do meu porto seguro./i,
    )

    expect(icon).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
