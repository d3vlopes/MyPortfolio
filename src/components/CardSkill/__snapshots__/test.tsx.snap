// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSkill /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #DEE7EA;
  border-radius: 1.2rem;
  width: min(29rem,100%);
  padding: 3rem;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  height: min(29.6rem,33rem);
}

.c0:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  background: #CAD5D9;
  box-shadow: 0px 10px 15px rgba(202,213,217,0.1);
}

.c0:hover .c2::after {
  content: '';
  position: absolute;
  left: 0;
  bottom: -0.8rem;
  margin-bottom: 1.6rem;
  width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border: 0.2rem solid #6D59A8;
  margin-bottom: 0.8rem;
}

.c1 {
  max-width: min(6rem,100%);
}

.c3 {
  position: relative;
  font-family: Gilroy-Bold;
  font-size: 2.2rem;
  color: #1A1C23;
  padding: 1.6rem 0 0.8rem 0;
}

.c4 {
  font-size: 1.8rem;
  line-height: 150%;
  text-align: center;
  color: #1A1C23;
  padding: 1.6rem;
}

<div
  class="c0"
>
  <img
    alt="Desafios"
    class="c1"
    src="/img/icons/sword.svg"
  />
  <h3
    class="c2 c3"
  >
    Desafios
  </h3>
  <p
    class="c4"
  >
    Estou sempre em busca de novos desafios que me tirem do meu porto seguro.
  </p>
</div>
`
